### PR TITLE
refactor: move config parsing to Radio

### PR DESCRIPTION
**Pull request by axiomatic-aardvark**
### Description

Moving the Config struct that is responsible for parsing env vars from the command line or env variables away from the SDK and into the Radio. Now the only way to pass these variables into the SDK is through GraphcastAgent::new.

Corresponding SDK PR - https://github.com/graphops/graphcast-sdk/issues/188

### Issue link (if applicable)
https://github.com/graphops/graphcast-sdk/issues/188

### Checklist

- [x] Have you tested your changes?
- [x] Does this PR include changes that require our documentation to be updated, if so, have you created a PR on the docs repo to make the necessary changes? - **this PR should not require docs changes**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -95,25 +92,28 @@ async fn main() {
     info!("Found content topics for subscription: {:?}", topics);
 
     debug!("Initializing Graphcast Agent");
+
+    let graphcast_agent_config = GraphcastAgentConfig {
```
Comment (hopeyen) [inline src/main.rs:96]: maybe there could be a radio Config function that converts the radio specific config to GraphcastAgentConfig? It's not necessary but would be cleaner?

File: src/main.rs [hunk 2]
```diff
@@ -62,10 +63,6 @@ async fn main() {
         ));
     }
 
-    if let Err(e) = config.validate_set_up().await {
-        panic!("Could not validate the supplied configurations: {e}")
-    }
-
     // Using unwrap directly as the query has been ran in the set-up validation
     let wallet = build_wallet(config.wallet_input().unwrap()).unwrap();
```
Comment (hopeyen) [inline src/main.rs:67]: This and following unwrap on config fields are no longer safe to do before configurations only gets validated after GraphcastAgentConfig is used for initializing GraphcastAgent. The users would lose the helpful logs from validating the setup. 

To retain the helpful error messages, maybe move initialization of GraphcastAgent (thus config validation up), or add setup validation to radio specific configs.

File: src/main.rs (hunk 1 above)
Comment (axiomatic-aardvark) [reply src/main.rs:96]: the Radio config could grow a lot and be more complex, I would really prefer to decouple them

File: src/main.rs (hunk 2 above)
Comment (axiomatic-aardvark) [reply src/main.rs:67]: great point, I'll probably move the GraphcastAgent init up

File: src/main.rs (hunk 1 above)
Comment (hopeyen) [reply src/main.rs:96]: my point is based on the thought that not all fields in Radio config should be `GraphcastAgentConfigs`, but Radio config must include all fields in `GraphcastAgentConfig`, such that whatever Radio config is defined, they should have a function that, convert might be the wrong word, but derive GraphcastAgentConfig from the Radio Config

**Review by hopeyen — APPROVED** [review]
🔥 🧙‍♂️

**Pull request merged**